lib/Service/PageService.php

```diff
@@ -902,13 +817,89 @@ public function delete(int $collectiveId, int $parentId, int $id, string $userId
 		} catch (FilesNotPermittedException $e) {
 			throw new NotPermittedException($e->getMessage(), 0, $e);
 		}
-		$this->pageMapper->deleteByFileId($pageInfo->getId());
-		$this->removeFromSubpageOrder($collectiveId, $parentId, $id, $userId);
 
-		$this->revertSubFolders($folder);
+		$this->initTrashBackend();
+		if (!$this->trashBackend) {
+			// Delete directly if trash is not available
+			$this->pageMapper->deleteByFileId($id);
+			$this->removeFromSubpageOrder($collectiveId, $parentId, $id, $userId);
+			NodeHelper::revertSubFolders($folder);
+			// TODO: check
+			return $pageInfo;
+		}
+
+		$this->notifyPush($collectiveId, $userId);
```

Comment:
Maybe move this above the `trashBackend` check so it also happens without a trash backend?

Reply:
Also a good catch. I solved this another way: Moved `notifyPush` to the very end so it only happens if the trash process was successful, but also added it to the the conditional code if `trashBackend` doesn't exist.